tensorboard/plugins/histogram/summary_v2.py

```diff
@@ -196,53 +196,50 @@ def _buckets(data, bucket_count=None):
         tf.debugging.assert_type(bucket_count, tf.int32)
         data = tf.reshape(data, shape=[-1])  # flatten
         data = tf.cast(data, tf.float64)
-        is_empty = tf.equal(tf.size(input=data), 0)
-
-        def when_empty():
+        data_size = tf.size(input=data)
+        if tf.equal(data_size, 0):
```

Comment:
So one issue here is that we can't really use normal Python control flow (it works in some contexts, like eager mode and like a tf.function with autograph enabled, but these functions also need to work in tf.functions without autograph, and in legacy graph mode - some context at https://www.tensorflow.org/guide/function#autograph_transformations).

Instead, we can use tf.cond() like the original code had in order to achieve the effect of an if statement.  Same goes for the indices below.